content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,367 @@
+Also each node is potentially a member of a group of nodes, that another considers a benchmarking group.
+On a higher level, the structure looks like multiple overlapping groups of participants.
+
+Eventually, when enough of the network accepts a transaction, it becomes infeasible and cannot be rolled back.
+This structure brings a high level of decentralization of decision making process among network members and mimics the behavior of social groups spreading a piece of information.
+
+Federated Byzantine Agreement is a model suitable for a worldwide consensus.
+One of the famous implementations of FBA is the [Stellar Consensus Protocol (SCP)](https://www.stellar.org/papers/stellar-consensus-protocol).
+Like non-federated Byzantine agreement, FBA addresses the problem of updating replicated state, such as transaction ledger or certificate tree and by agreeing on what updates should be applied to the state, nodes avoid contradictory values and irreconcilable states.
+
+Each state update is identified by a unique "slot", from which inter-update dependencies can be inferred.
+For example slots may be consecutively numbered positions in a sequentially applied log.
+FBA system runs a consensus protocol that ensures nodes are agreeing on slot contents.
+A node can update a value `x` in slot `i` when it has applied updates in all slots upon which `i` depends, additionally — the node believes that all correctly functioning nodes will eventually agree on value `x` for slot `i`.
+This action is called externalization — the node externalized value `x` for slot `i`.
+Outside world reacts on the externalized value — agrees or objects it.
+
+#### Quorum slices
+
+As a Byzantine agreement system, FBA uses the notion of `quorum` once again as a form of a majority rule.
+In an FBA consensus protocol implementation, nodes exchange messages asserting statements about slots.
+When a node hears a sufficient set of nodes assert one statement, it assumes no functioning node will ever contradict that statement.
+This sufficient set of nodes is called a `quorum slice` or just a `slice`.
+A node in an FBA network refuses to commit to a particular value until it sees its peers are also committing to it.
+
+Since in FBA the collections of participants are loosely defined, nodes quorums are not a set or a subset of a predetermined roster of nodes, but are dynamically defined from an ever-changing and inevitably incomplete snapshot of membership.
+Each node has multiple slices, any of which is sufficient to convince it of a value, so we can say that an FBA system consists of a loose federation of nodes each of which has chosen one or more slices to serve as benchmarking for making decisions.
+
+Forming a quorum starts with a forming a quorum slice.
+A node adds to its quorum slice nodes it knows about as well as their peers — other nodes they know about.
```

Comment:
`adds other nodes to its quorum slice`

Reply:
This has been rephrased https://github.com/optakt/optakt.github.io/pull/21/commits/ca80e6c74bee64b8ce712ca515bb205ee645d971#diff-4bf8323c6426a93da1ce22be1b8528779ef6b61fdacef3e0d065f1e1dd47bea5R92